src/box/index.h

```diff
@@ -375,48 +375,39 @@ iterator_next(struct iterator *it, struct tuple **ret);
 int
 iterator_next_internal(struct iterator *it, struct tuple **ret);
 
-/*
- * Iterator position has complex format in MsgPack:
- * +--------+--------+--------------+========================+
- * | MP_BIN | MP_MAP | POSITION_KEY | KEY IN MP_ARRAY FORMAT |
- * +--------+--------+--------------+========================+
- * MP_BIN - needed to make the object opaque to users working
- * directly with IPROTO.
- * MP_MAP - needed for extensibility (at least, we have an idea
- * to generate a digital signature to make sure that user did
- * not modify the object).
- * All the keys of map should be unsigned integer values to
- * minimize the size of the object.
- *
- */
-
 /** Calculate length of packed iterator position. */
 uint32_t
-iterator_position_pack_size(const char *pos, const char *pos_end,
-			    uint32_t part_count);
+iterator_position_pack_size(const char *pos, const char *pos_end);
 
 /**
- * Pack iterator position to preallocated buffer. Buffer length must be
- * more or equal than packed size of position.
+ * Pack (encode to base64 format) iterator position with MP_ARRAY header to
+ * preallocated buffer. Buffer length must be more or equal than packed size
+ * of position.
  */
 void
 iterator_position_pack(const char *pos, const char *pos_end,
-		       uint32_t part_count, char *packed_pos,
-		       const char *packed_pos_end);
+		       char *packed_pos, const char *packed_pos_end);
+
+/**
+ * Maximum possible size of unpacked iterator position.
+ */
+uint32_t
+iterator_position_unpack_size_max(const char *packed_pos,
+				  const char *packed_pos_end);
 
 /**
- * Unpack iterator position descriptor from MsgPack. Does not allocate memory,
- * so lifetime of returned position is the same as lifetime of passed ptr.
+ * Unpack iterator position descriptor from MsgPack to preallocated buffer.
+ * Returned position is a MsgPack key without MP_ARRAY header.
```

Comment:
I don't like this inconsistency between pack and unpack. Let's return the value *with* `MP_ARRAY` header and strip it in `box_select`.

Reply:
Done